docs/modelserving/v1beta1/llm/huggingface/README.md

```diff
@@ -0,0 +1,159 @@
+      transformer:
+        containers:
+        - args:
+          - --model_name=llama2
+          - --model_id=meta-llama/Llama-2-7b-chat-hf
+          - --predictor_protocol=v2
+          - --tensor_input_names=input_ids
+          - --disable_vllm
+          image: kserve/huggingfaceserver:latest
+          name: kserve-container
+          resources:
+            limits:
+              cpu: "1"
+              memory: 2Gi
+            requests:
+              cpu: 100m
+              memory: 2Gi
+    EOF
+    ```
+
+### Serve the huggingface model using vllm runtime
+
+=== "Yaml"
+    ```yaml
+    kubectl apply -f - <<EOF
+    apiVersion: serving.kserve.io/v1beta1
+    kind: InferenceService
+    metadata:
+      name: huggingface
+    spec:
+      predictor:
+        containers:
+        - args:
+          - --model_name=llama2
+          - --model_id=meta-llama/Llama-2-7b-chat-hf
+          image: kserve/huggingfaceserver:latest
+          name: kserve-container
+          resources:
+            limits:
+              cpu: "6"
+              memory: 24Gi
+              nvidia.com/gpu: "1"
+            requests:
+              cpu: "6"
+              memory: 24Gi
+              nvidia.com/gpu: "1"
+    EOF
+    ```
+
+### Model Inference
+
+The first step is to [determine the ingress IP and ports](../../../../get_started/first_isvc.md#4-determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`.
+
+```bash
+MODEL_NAME=llama2
+SERVICE_HOSTNAME=$(kubectl get inferenceservice huggingface -o jsonpath='{.status.url}' | cut -d "/" -f 3)
+```
+
+Inference KServe python runtime with v1 REST Protocol
+
+```bash
+curl -H "content-type:application/json" -H "Host: ${SERVICE_HOSTNAME}" -v http://${INGRESS_HOST}:${INGRESS_PORT}/v1/models/${MODEL_NAME}:predict -d '{"instances": ["The capital of france is [MASK]."] }'
+
+```
+!!! success "Expected Output"
+
+  ```{ .bash .no-copy }
+    {"predictions":["The capital of France is actually Paris."]}
+  ```
+
+Inference triton runtime with v2 REST Protocol
+
+```bash
+curl -H "content-type:application/json" -H "Host: ${SERVICE_HOSTNAME}" -v http://${INGRESS_HOST}:${INGRESS_PORT}/v2/models/${MODEL_NAME}/infer -d '{"id": "42","inputs": [{"name": "input0","shape": [-1],"datatype": "BYTES","data": ["The capital of france is [MASK]."]}]}'
+
+```
+!!! success "Expected Output"
+
+  ```{ .bash .no-copy }
+    {"model_name":"llama2", "model_version":null, "id":42, "outputs":[{"name":predict, "shape":[-1], "datatype":"BYTES", "data":["The capital of France is actually Paris."]}]}
+  ```
+
+vLLM runtime only supports the [/generate](https://github.com/kserve/open-inference-protocol/blob/main/specification/protocol/generate_rest.yaml) endpoint schema for inference.
+
+```bash
+curl -H "content-type:application/json" -H "Host: ${SERVICE_HOSTNAME}" -v http://${INGRESS_HOST}:${INGRESS_PORT}/v2/models/${MODEL_NAME}/generate -d '{"text_input": "The capital of france is [MASK]." }'
```

Comment:
This is an BERT example for mask filling task, we should use an input for text generation case.